### ISSUE TITLE
Stage 2 training issues

### ISSUE THREAD

**Issue by Fixed-Sidet** [Thinklab-SJTU/Bench2DriveZoo#53]
Hello,

first, thanks for your outstanding work!

I am curently trying to train UniAD on the Carla dataset.
Due to ressource limitations, I reduced the size of the training dataset to 150 scenes, while keeping the distribution of towns and weather conditions equal.

However, when training stage2 I run into the following problem:
![overall_loss](https://github.com/user-attachments/assets/c8802749-b6ac-427c-af38-35698c6d692c)
The loss goes up and then drops down again after a while.
This can be seen most clearly in the past_traj loss:

![pic](https://github.com/user-attachments/assets/7579f270-dc3a-4a9e-8a13-6e468d1cd206)

it is however also present in the motion.ade loss as well as others.
Removing the past_traj loss from the training does not solve the issue.

Maybe you can give me an insight in what the problem is here.

Thanks!

**Comment by jiaxiaosong1002** [comment]
@Fixed-Sidet It might be due to the sudden disappear of some vehicles in CARLA https://github.com/Thinklab-SJTU/Bench2Drive/issues/31, while the TrackFormer could not handle it well.
